File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    arr.sort((a,b) => b - a)
    let r, m, n = 0, acc = arr[0]
    for (let i = 1; i < arr.length; ++i) {
        m = acc
        n = arr[i]
        while (0 < n) {
            r = m % n
            m = n
            n = r
        }
        acc = acc * arr[i] / m
    }
    return acc;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
import math

def solution(arr):
    arr.sort(reverse=True)
    
    if not arr:
        return 0

    acc = arr[0]

    for i in range(1, len(arr)):
        m_val = acc
        n_val = arr[i]
        gcd_val = math.gcd(m_val, n_val)
        acc = (acc * arr[i]) // gcd_val
    return acc